### ISSUE TITLE
--no-checked is ignored? Still seeing SLOW MODE banner

### ISSUE THREAD

**Issue by sethladd** [flutter/flutter#3850]
Using latest master. I used `--no-checked` but I still see a SLOW MODE banner.

```
~/Code/flutter/examples/flutter_gallery[master*] $ flutter -d 01abfc49119c410e run --no-checked
Building APK in debug mode (android-arm)...
Warning: signing the APK using the debug keystore.
Built build/app.apk (19.7MB).
Installing app.apk to Nexus 5X...
Running lib/main.dart on Nexus 5X...
Observatory listening on http://127.0.0.1:8100
Diagnostic server listening on http://127.0.0.1:8101
```

Targeting a Nexus 5X, building from a Mac.

**Comment by devoncarew** [comment]
`--no-checked` (and `--checked`) are now no-op options, as of https://github.com/flutter/flutter/pull/3799

I'm hoping we can get away with fewer flags, and that `--debug`, `--profile`, and `--release` will be sufficient to cover most use cases.

This means that you can't debug with checked mode off. Would `--profile` or `--release` work for what you're doing?

**Comment by sethladd** [comment]
I wonder if we want to print something like "--no-checked no longer does anything. Use --debug, --profile, or --release instead. Visit _shortlink_ to learn more."

tldr: on a mac, there doesn't seem to be a way to build for non-slow-mode ?  (I know this is coming soon, but not sure if there's a workaround in the meantime)

**Comment by devoncarew** [comment]
There isn't a workaround in the meantime; if it's blocking you I can re-enable the checked flag and remove it again once the three modes are further along.

**Comment by sethladd** [comment]
I needed to make a non-slow-mode build for a review. Suggest either keeping --no-checked behavior until the Mac has a way to do --release, or print something when --no-checked is used, pointing the user to https://github.com/flutter/flutter/wiki/Flutter's-modes (or similar)

**Comment by sethladd** [comment]
Thanks for the temp. revert. I think that closing https://github.com/flutter/flutter/issues/3269 will allow us to remove --no-checked on Mac.
